internals/secrethub/credential_backup_test.go

```diff
@@ -0,0 +1,128 @@
+	var testError = errors.New("test")
+
+	testCases := map[string]struct {
+		cmd            CredentialBackupCommand
+		promptOut      string
+		out            string
+		in             string
+		username       string
+		createError    error
+		meError        error
+		newClientError error
+		promptError    error
+		err            error
+	}{
+		"fail-client-error": {
+			err:            testError,
+			newClientError: testError,
+		},
+		"fail-me-error": {
+			err:     testError,
+			meError: testError,
+		},
+		"fail-abort": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in:  "n",
+			out: "Aborting\n",
+		},
+		"fail-create-error": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in:          "y",
+			err:         testError,
+			createError: testError,
+		},
+		"fail-no-account-key": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in:  "y",
+			err: errors.New("backup code has not yet been generated"),
+		},
+
+		"success": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in: "y",
+			out: "This is your backup code: \n%s\n" + "Write it down and store it in a safe location! " +
+				"You can restore your account by running `secrethub init`.",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			io := fakeui.NewIO(t)
+			io.PromptIn.Buffer = bytes.NewBufferString(tc.in)
+			io.PromptErr = tc.promptError
+			tc.cmd.io = io
+
+			tc.cmd.newClient = func() (secrethub.ClientInterface, error) {
+				client := fakeclient.Client{
+					MeService: &fakeclient.MeService{
+						GetUserFunc: func() (*api.User, error) {
+							return &api.User{
+								Username: "Chucky",
+							}, tc.meError
+						},
+					},
+					CredentialService: &fakeclient.CredentialService{
+						CreateFunc: func(creator credentials.Creator, s string) (*api.Credential, error) {
+							if name == "success" {
+								_ = creator.Create()
+							}
+							return &api.Credential{}, tc.createError
+						},
+					},
+				}
+				return client, tc.newClientError
+			}
+
+			err := tc.cmd.Run()
+
+			assert.Equal(t, err, tc.err)
+
+			// Since at the moment there is no way to retrieve the generated backup code
+			// we should just compare the beginning and the end of the message with the
+			// expected values, omitting, in this way, any assertion on the backup code.
```

Comment:
Can you please explain how would I do these assertions?
Since the secrethub assertion package does not provide any functionality resembling "assertTrue"?

Reply:
Sure! You can also use the [`t.Error](https://golang.org/pkg/testing/#T.Error) (or `t.Errorf`) in an if statement, to make the test fail with a clear error message.

Something like:
```go
if !matches {
    t.Errorf("The output did not match the expected format. Got: %s, but expected string of format: %s")
}
```

Oh, and maybe it would be even more readable if you used a regular expression like here: https://github.com/secrethub/secrethub-go/blob/9cc951048891ec17e56d4130cba3bf24fe9c1257/pkg/secrethub/client_test.go#L73

But I'm not saying you should. Do whatever makes more sense to you.